include/circt/Dialect/DC/DCOps.td

```diff
@@ -0,0 +1,395 @@
+//===- DCOps.td - DC dialect operations --------------------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef CIRCT_DIALECT_DC_OPS_TD
+#define CIRCT_DIALECT_DC_OPS_TD
+
+include "circt/Dialect/DC/DCDialect.td"
+include "circt/Dialect/DC/DCTypes.td"
+include "mlir/IR/FunctionInterfaces.td"
+include "mlir/IR/OpAsmInterface.td"
+include "mlir/IR/RegionKindInterface.td"
+include "mlir/IR/SymbolInterfaces.td"
+include "mlir/Interfaces/CallInterfaces.td"
+include "mlir/Interfaces/ControlFlowInterfaces.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+include "mlir/IR/BuiltinAttributes.td"
+include "mlir/IR/FunctionInterfaces.td"
+
+class DCOp<string mnemonic, list<Trait> traits = []> :
+  Op<DCDialect, mnemonic, !listconcat(traits, [
+    Pure
+  ])>;
+
+class SameTypeConstraint<string lhs, string rhs>
+ : TypesMatchWith<"lhs and rhs types should be equivalent",
+                   lhs, rhs, [{ $_self }]>;
+
+def FuncOp : Op<DCDialect, "func", [
+      IsolatedFromAbove,
+      Symbol,
+      FunctionOpInterface,
+      RegionKindInterface,
+      CallableOpInterface,
+  ]> {
+  // I really don't want to have to add yet another container operation. However,
+  // where else do we place DC ops?
+  // We have to place them in a graph region operation, so func.func isn't
+  // applicable. We're not committed to being in `hw` yet, so hw.module doesn't
```

Comment:
If I understand the DC dialect properly, these ops can go in any graph container block. You just have to select one (or several) purely for testing and only in the test files. In which case `hw.module` is appropriate. The `msft.module` op could also be used.

So I'm going to ignore this and the next few ops.